pkg/builder/builder.go

```diff
@@ -32,6 +32,10 @@ import (
 	"k8s.io/kube-state-metrics/v2/pkg/options"
 )
 
+// Make sure the public Builder implements the public BuilderInterface.
+// New internal Builder methods should be added to the public BuilderInterface.
+var _ ksmtypes.BuilderInterface = &Builder{}
```

Comment:
Can we move this to builder_test.go ?

Reply:
we can.
I added it the `builder.go` to get quick feedback at build time

Reply:
Looks like we have the same in pkg/internal/store/builder.go so I guess it should be fine here to keep things consistent.